ui/app/src/utils/itemActions.ts

```diff
@@ -819,6 +819,44 @@ export const itemActionDispatcher = ({
       );
       break;
     }
+    case 'deleteController': {
+      const id = uuid();
+      dispatch(
+        pushDialog({
+          minimized: true,
+          id,
+          status: 'indeterminate',
+          title: 'Processing', // TODO: i18n?
+          onMaximized: null
+        })
+      );
+      const currentItem = item as DetailedItem;
+      const contentTypeName = /[^/]*$/.exec(currentItem.contentTypeId)[0];
+      fetchSandboxItem(site, `/scripts/${currentItem.systemType}s/${contentTypeName}.groovy`).subscribe(
+        (controllerItem) => {
+          dispatch(popDialog({ id }));
+
+          let items = asArray(controllerItem);
+          dispatch(
+            showDeleteDialog({
+              items,
+              onSuccess: batchActions([
+                showDeleteItemSuccessNotification(),
+                closeDeleteDialog(),
+                ...(onActionSuccess ? [onActionSuccess] : [])
+              ])
+            })
+          );
+        },
+        (e) => {
+          dispatch(popDialog({ id }));
+        }
+      );
+      break;
+    }
+    case 'deleteTemplate': {
+      break;
```

Comment:
Once a template is deleted, we must update the content type so the content type template is nulled.

Reply:
use & test `dissociateTemplate` service